File: projectBlueWater/src/main/java/com/lasthopesoftware/bluewater/client/connection/IConnectionProvider.java
package com.lasthopesoftware.bluewater.client.connection;

import com.lasthopesoftware.bluewater.client.connection.url.IUrlProvider;
import com.namehillsoftware.handoff.promises.Promise;
import okhttp3.Response;

public interface IConnectionProvider {
	Promise<Response> promiseResponse(String... params);

	IUrlProvider getUrlProvider();
}

File: projectBlueWater/src/main/java/com/lasthopesoftware/bluewater/client/connection/IConnectionProvider.kt
package com.lasthopesoftware.bluewater.client.connection

import com.lasthopesoftware.bluewater.client.connection.url.IUrlProvider
import com.namehillsoftware.handoff.promises.Promise
import okhttp3.Response

interface IConnectionProvider {
	fun promiseResponse(vararg params: String): Promise<Response>
	fun promiseSentPacket(packet: ByteArray): Promise<Unit>
	val urlProvider: IUrlProvider
}
